flyteidl/protos/flyteidl/plugins/ray.proto

```diff
@@ -37,8 +37,8 @@ message HeadGroupSpec {
   // Optional. RayStartParams are the params of the start command: address, object-store-memory.
   // Refer to https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-start
   map<string, string> ray_start_params = 1;
-  // Resource specification for ray head pod
-  core.Resources resources = 2;
+  // Pod Spec for the ray head pod
+  core.K8sPod k8s_pod = 2;
```

Comment:
let's not reuse field numbers. Mark the previous field as deprecated and leave a comment so that IDEs show that explicitly (e.g. https://github.com/flyteorg/flyte/blob/master/flyteidl/protos/flyteidl/admin/task_execution.proto#L79-L81)

Reply:
These are net new fields, if you mean it reused between the two commits in the pull request I'll just squash my changes into a single commit.

Reply:
Fixed